analysis/src/Packages.ml

```diff
@@ -44,165 +44,167 @@ let newBsPackage ~rootPath =
-                      intModulePath = ["Int"];
-                      floatModulePath = ["Float"];
-                      promiseModulePath = ["Js"; "Promise"];
-                      listModulePath = ["List"];
-                      resultModulePath = ["Result"];
-                      exnModulePath = ["Js"; "Exn"];
-                      regexpModulePath = ["Js"; "Re"];
-                    }
-                  else
-                    {
-                      arrayModulePath = ["Js"; "Array2"];
-                      optionModulePath = ["Belt"; "Option"];
-                      stringModulePath = ["Js"; "String2"];
-                      intModulePath = ["Belt"; "Int"];
-                      floatModulePath = ["Belt"; "Float"];
-                      promiseModulePath = ["Js"; "Promise"];
-                      listModulePath = ["Belt"; "List"];
-                      resultModulePath = ["Belt"; "Result"];
-                      exnModulePath = ["Js"; "Exn"];
-                      regexpModulePath = ["Js"; "Re"];
-                    });
-               uncurried;
-             })))
+      in
+      let uncurried = uncurried = Some true in
+      match libBs with
+      | None -> None
+      | Some libBs ->
+        let cached = Cache.readCache (Cache.targetFileFromLibBs libBs) in
+        let projectFiles, dependenciesFiles, pathsForModule =
+          match cached with
+          | Some cached ->
+            ( cached.projectFiles,
+              cached.dependenciesFiles,
+              cached.pathsForModule )
+          | None ->
+            let dependenciesFilesAndPaths =
+              match FindFiles.findDependencyFiles rootPath config with
+              | None -> []
+              | Some (_dependencyDirectories, dependenciesFilesAndPaths) ->
+                dependenciesFilesAndPaths
+            in
+            let sourceDirectories =
+              FindFiles.getSourceDirectories ~includeDev:true ~baseDir:rootPath
+                config
+            in
+            let projectFilesAndPaths =
+              FindFiles.findProjectFiles
+                ~public:(FindFiles.getPublic config)
+                ~namespace ~path:rootPath ~sourceDirectories ~libBs
+            in
+            let pathsForModule =
+              makePathsForModule ~projectFilesAndPaths
+                ~dependenciesFilesAndPaths
+            in
+            let projectFiles =
+              projectFilesAndPaths |> List.map fst |> FileSet.of_list
+            in
+            let dependenciesFiles =
+              dependenciesFilesAndPaths |> List.map fst |> FileSet.of_list
+            in
+            (projectFiles, dependenciesFiles, pathsForModule)
+        in
+        Some
+          (let opens_from_namespace =
+             match namespace with
+             | None -> []
+             | Some namespace ->
+               let cmt = Filename.concat libBs namespace ^ ".cmt" in
+               Hashtbl.add pathsForModule namespace (Namespace {cmt});
```

Comment:
When the cache file is read, `pathsForModule` will already have been populated when this line was executed when creating the cache, and here it is added again.
No consequence I can think of, but good practice to make sure things are not mutated after caching them.

Reply:
This should be fine because this is in the non-cached case, right?

Reply:
Good catch. Let me think about how to make that better.

Reply:
Maybe it's fine to leave in for now. I can't think of a consequence either. Maybe doing a Hashtbl.replace instead of add makes sense though.